src/mod/applications/mod_callcenter/mod_callcenter.c

```diff
@@ -40,6 +40,7 @@
 #define CC_SQLITE_DB_NAME "callcenter"
 #define CC_APP_KEY "mod_callcenter"
 
+#define CC_OFFERED_AGENT_LIST_SIZE 2048
```

Comment:
why 2048?

Reply:
@wowi42 
this is used to maintain a string array of agent names at each level that will be used to ignore after they are offered once so, the ringing call can be offered to other available agents at this level.

example -> agents available at level 1:  cclogin1@default, cclogin2@default, cclogin3@default

say call comes and randomly offered to agent -> cclogin2@default, if he does not pick then call will go to either of the left agents and this agent will go to ignore list in below query to pick next available agent - 

query to fetch available agent -> 
`SELECT instance_id, name, status, contact, no_answer_count, max_no_answer, reject_delay_time, busy_delay_time, no_answer_delay_time, tiers.state, agents.last_bridge_end, agents.wrap_up_time, agents.state, agents.ready_time, tiers.position as tiers_position, tiers.level as tiers_level, agents.type, agents.uuid, external_calls_count, agents.last_offered_call as agents_last_offered_call FROM agents LEFT JOIN tiers ON (agents.name = tiers.agent) WHERE tiers.queue = 'abraKaDabra@default' AND (agents.status = 'Available' OR agents.status = 'On Break' OR agents.status = 'Available (On Demand)') AND tiers.level >= 2 AND name NOT IN ('0002@default') ORDER BY tiers_level ASC, random(), agents_last_offered_call`


the max an agent name can go is -> 18 characters ('000000@default')
so at each level, we can server ~110 agents in the ignore list